docs/distributed/distributed_autofaiss.md

```diff
@@ -0,0 +1,222 @@
+# distributed autofaiss
+
+If you want to generate an index from billion of embeddings, this guide is for you.
+
+This guide is about using pyspark to run autofaiss in multiple nodes.
+
+you may also be interested by [distributed img2dataset](https://github.com/rom1504/img2dataset/blob/main/examples/distributed_img2dataset_tutorial.md)
+and [distributed clip inference](https://github.com/rom1504/clip-retrieval/blob/main/docs/distributed_clip_inference.md)
+
+We will be assuming ubuntu 20.04.
+
+## Setup the master node
+
+On the master node:
+
+First download spark:
+```bash
+wget https://archive.apache.org/dist/spark/spark-3.2.1/spark-3.2.1-bin-hadoop3.2.tgz
+tar xf spark-3.2.1-bin-hadoop3.2.tgz
+```
+
+Then download autofaiss:
+```bash
+rm -rf autofaiss.pex
+wget https://github.com/rom1504/autofaiss/releases/latest/download/autofaiss-3.8.pex -O autofaiss.pex
+chmod +x autofaiss.pex
+```
+
+If the master node cannot open ports that are visible from your local machine, you can do a tunnel between your local machine and the master node to be able to see the spark ui (at http://localhost:8080)
+```bash
+ssh -L 8080:localhost:8080 -L 4040:localhost:4040 master_node
```

Comment:
Question: the `master_node` here is just a label or address ?

Reply:
It's an address, will clarify

Reply:
fixed